src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.RateLimiting;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+/// <summary>
+/// Extension methods for the RateLimiting middleware options.
+/// </summary>
+public static class RateLimiterOptionsExtensions
+{
+    /// <summary>
+    /// Adds a new <see cref="TokenBucketRateLimiter"/> with the given <see cref="TokenBucketRateLimiterOptions"/> to the <see cref="RateLimiterOptions"/>.
+    /// </summary>
+    /// <param name="options">The <see cref="RateLimiterOptions"/> to add a limiter to.</param>
+    /// <param name="policyName">The name that will be associated with the limiter.</param>
+    /// <param name="tokenBucketRateLimiterOptions">The <see cref="TokenBucketRateLimiterOptions"/> to be used for the limiter.</param>
+    /// <returns>This <see cref="RateLimiterOptions"/>.</returns>
+    public static RateLimiterOptions AddTokenBucketLimiter(this RateLimiterOptions options, string policyName, TokenBucketRateLimiterOptions tokenBucketRateLimiterOptions)
+    {
+        var key = new PolicyNameKey() { PolicyName = policyName };
+        return options.AddPolicy(policyName, context =>
+        {
+            return RateLimitPartition.CreateTokenBucketLimiter(key,
+                _ => tokenBucketRateLimiterOptions);
+        });
+    }
+
+    /// <summary>
+    /// Adds a new <see cref="FixedWindowRateLimiter"/> with the given <see cref="FixedWindowRateLimiterOptions"/> to the <see cref="RateLimiterOptions"/>.
+    /// </summary>
+    /// <param name="options">The <see cref="RateLimiterOptions"/> to add a limiter to.</param>
+    /// <param name="policyName">The name that will be associated with the limiter.</param>
+    /// <param name="fixedWindowRateLimiterOptions">The <see cref="FixedWindowRateLimiterOptions"/> to be used for the limiter.</param>
+    /// <returns>This <see cref="RateLimiterOptions"/>.</returns>
+    public static RateLimiterOptions AddFixedWindowLimiter(this RateLimiterOptions options, string policyName, FixedWindowRateLimiterOptions fixedWindowRateLimiterOptions)
```

Comment:
Should there also be an API to accept `Func<HttpContext, FixedWindowRateLimiterOptions> partitioner`?

Reply:
There's such a method on the `RateLimiterOptions` - one of the `AddPolicy` overloads

Reply:
If I understand it correctly you mean like this?
```csharp
options.AddPolicy("policyName", context => RateLimitPartition.CreateFixedWindowLimiter<HttpContext>(partitionkey, context => new FixedWindowRateLimiterOptions());
```
but perhaps this might be more discoverable.
```csharp
options.AddFixedWindowLimiter("policyName", context => new FixedWindowRateLimiterOptions());
```